File: Week 1 (C)/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    int cents;
    do
    {
        cents = get_int("Changed owed: ");
    } while (cents < 0);
    return cents;
}

int calculate_quarters(int cents)
{
    int x = 0;
    if (cents >= 25)
    {
        for (int i = 25; i <= cents; i += 25)
        {
            x = x + 1;
        }
    }
    return x;
}

int calculate_dimes(int cents)
{
    int x = 0;
    if (cents >= 10)
    {
        for (int i = 10; i <= cents; i += 10)
        {
            x = x + 1;
        }
    }
    return x;
}

int calculate_nickels(int cents)
{
    int x = 0;
    if (cents >= 5)
    {
        for (int i = 5; i <= cents; i += 5)
        {
            x = x + 1;
        }
    }
    return x;
}

int calculate_pennies(int cents)
{
    int x = 0;
    if (cents >= 1)
    {
        for (int i = 1; i <= cents; i += 1)
        {
            x = x + 1;
        }
    }
    return x;
}

File: Week 6 (Python)/sentimental-cash/cash.py
# TODO
from cs50 import *


def get_cents():
    while True:
        cents = get_float("Changed owed: ")
        if cents > 0:
            break
    return int(cents * 100)


def calculate_quarters(cents):
    x = 0
    if cents >= 25:
        for i in range(25, cents + 1, 25):
            x += 1
    return x


def calculate_dimes(cents):
    x = 0
    if cents >= 10:
        for i in range(10, cents + 1, 10):
            x += 1
    return x


def calculate_nickels(cents):
    x = 0
    if cents >= 5:
        for i in range(5, cents + 1, 5):
            x += 1
    return x


def calculate_pennies(cents):
    x = 0
    if cents >= 1:
        for i in range(1, cents + 1, 1):
            x += 1
    return x


def main():
    cents = get_cents()

    quarters = int(calculate_quarters(cents))
    cents -= quarters * 25

    dimes = int(calculate_dimes(cents))
    cents -= dimes * 10

    nickels = int(calculate_nickels(cents))
    cents -= nickels * 5

    pennies = int(calculate_pennies(cents))
    cents -= pennies * 1

    coins = quarters + dimes + nickels + pennies

    print(coins)


main()
